libraries/Microsoft.Bot.Builder.Core.Extensions/IStorage.cs

```diff
@@ -48,22 +48,19 @@ public static class StorageExtensions
         /// <param name="storage"></param>
         /// <param name="keys"></param>
         /// <returns></returns>
-        public static async Task<IEnumerable<KeyValuePair<string, StoreItemT>>> Read<StoreItemT>(this IStorage storage, params string[] keys) where StoreItemT : class
+        public static async Task<IDictionary<string, StoreItemT>> Read<StoreItemT>(this IStorage storage, params string[] keys) where StoreItemT : class
         {
             var storeItems = await storage.Read(keys).ConfigureAwait(false);
-
-            return ReturnStoreItemsOfDesiredType();
-
-            IEnumerable<KeyValuePair<string, StoreItemT>> ReturnStoreItemsOfDesiredType()
+            IDictionary<string, StoreItemT> values = new Dictionary<string, StoreItemT>();
+            foreach (var entry in storeItems)
             {
-                foreach (var entry in storeItems)
+                if (entry.Value is StoreItemT valueAsType)
                 {
-                    if (entry.Value is StoreItemT valueAsType)
-                    {
-                        yield return new KeyValuePair<string, StoreItemT>(entry.Key, valueAsType);
-                    }
+                    values.Add(new KeyValuePair<string, StoreItemT>(entry.Key, valueAsType));
```

Comment:
Would just use the `IDictionary<K,V>` friendly overload of `Add` here: `values.Add(entry.Key, valueAsType)`

Reply:
Not sure what I was thinking here. Thanks for the second set of eyes.